pkg/plugins/plugins.go

```diff
@@ -130,10 +128,18 @@ func ReadPluginJSON(reader io.Reader) (JSONData, error) {
 	switch plugin.ID {
 	case "grafana-piechart-panel":
 		plugin.Name = "Pie Chart (old)"
-	case "grafana-pyroscope-datasource": // rebranding
-		plugin.Alias = "phlare"
-	case "debug": // panel plugin used for testing
-		plugin.Alias = "debugX"
+	case "grafana-pyroscope-datasource":
+		fallthrough
+	case "annolist":
+		fallthrough
+	case "debug":
+		if plugin.Alias == "" {
+			return plugin, fmt.Errorf("expected alias to be set")
+		}
```

Comment:
this custom check seems to be not needed?

Reply:
i have no trouble removing it -- it just verifies that the alias was set the same from when we hardcoded it.

Reply:
yes, my concern is that people (us) start adding custom logic here to do more things (that's also why I like the idea of just having a map with the aliases)

Reply:
once we trust gcom... this whole section can be removed

Reply:
I still doubt we can remove it completely, unless we want to introduce a breaking change for offline installations.